app/src/main/java/com/mpieterse/stride/utils/ImageUploadHelper.kt

```
@@ -0,0 +1,89 @@
+            // Create temp file inside try block to handle creation errors consistently
+            tempFile = File.createTempFile("habit-", extension, context.cacheDir)
+            val bytes = Base64.decode(base64, Base64.DEFAULT)
+            tempFile.writeBytes(bytes)
+            when (val result = uploadRepo.upload(tempFile.path)) {
+                is ApiResult.Ok -> result.data.localPath
+                is ApiResult.Err -> {
+                    Clogger.e(TAG, "Image upload failed: ${result.message}")
+                    null
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)